pkg/machinery/cel/cel.go

```diff
@@ -0,0 +1,159 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package cel provides helpers for working with CEL expressions.
+package cel
+
+import (
+	"encoding"
+	"fmt"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/siderolabs/go-pointer"
+	"github.com/siderolabs/protoenc"
+	exprpb "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+	"gopkg.in/yaml.v3"
+
+	"github.com/siderolabs/talos/pkg/machinery/proto"
+)
+
+// Expression is a CEL expression that can be marshaled/unmarshaled as part of the resource.
+type Expression struct {
+	ast        *cel.Ast
+	expression *string
+}
+
+// Check interfaces.
+var (
+	_ encoding.TextMarshaler   = Expression{}
+	_ encoding.TextUnmarshaler = (*Expression)(nil)
+	_ yaml.IsZeroer            = Expression{}
+)
+
+// MustExpression panics if the expression cannot be parsed.
+func MustExpression(expr Expression, err error) Expression {
+	if err != nil {
```

Comment:
just checking error, no parsing?

Reply:
this is a wrapper to parse well-known valid expressions (static), not user input of course